apps/web/components/dashboard/BulkBookmarksAction.tsx

```diff
@@ -0,0 +1,204 @@
+"use client";
+
+import React, { useEffect, useState } from "react";
+import { usePathname } from "next/navigation";
+import { Button, ButtonWithTooltip } from "@/components/ui/button";
+import {
+  Dialog,
+  DialogClose,
+  DialogContent,
+  DialogFooter,
+  DialogHeader,
+  DialogTitle,
+} from "@/components/ui/dialog";
+import { useToast } from "@/components/ui/use-toast";
+import bulkActions from "@/store/bulkBookmarksAction";
+import { Archive, ArchiveRestore, Pencil, Star, Trash2, X } from "lucide-react";
+
+import {
+  useDeleteBookmark,
+  useUpdateBookmark,
+} from "@hoarder/shared-react/hooks/bookmarks";
+
+function DeleteModal({
+  open,
+  setOpen,
+  deleteBookmarks,
+}: {
+  open: boolean;
+  setOpen: (open: boolean) => void;
+  deleteBookmarks: () => void;
+}) {
+  return (
+    <Dialog open={open} onOpenChange={setOpen}>
+      <DialogContent>
+        <DialogHeader>
+          <DialogTitle>Delete Bookmarks</DialogTitle>
+          <DialogClose />
+        </DialogHeader>
+        <p>Are you sure you want to delete these bookmarks?</p>
+        <DialogFooter>
+          <Button variant="ghost" onClick={() => setOpen(false)}>
+            Cancel
+          </Button>
+          <Button
+            variant="destructive"
+            onClick={() => {
+              deleteBookmarks();
+              setOpen(false);
+            }}
+          >
+            Delete
+          </Button>
+        </DialogFooter>
+      </DialogContent>
+    </Dialog>
+  );
+}
+
+export default function BulkBookmarksAction() {
+  const { selectedBookmarks, isBulkEditEnabled } = bulkActions();
+  const handleBulkEdit = bulkActions((state) => state.handleBulkEdit);
+  const { toast } = useToast();
+  const pathname = usePathname();
+  const [showDeleteModal, setShowDeleteModal] = useState(false);
+
+  useEffect(() => {
+    handleBulkEdit(false); // turn off toggle + clear selected bookmarks on mount
+  }, []);
+
+  const onError = () => {
+    toast({
+      variant: "destructive",
+      title: "Something went wrong",
```

Comment:
this has been remarked before: People want to know more information. Can we provide information about the affected bookmark somehow?

Reply:
Let's pick this in separate PR
cc - @MohamedBassem